From e73805a78a8001203f866aeb24a56e1c1de36b50 Mon Sep 17 00:00:00 2001
From: Hao Peng <penghao@linuxdeepin.com>
Date: Thu, 5 Mar 2015 12:00:49 +0800
Subject: winecfg: Double click in dlls list to edit item's overides. (try 3)

Add a feature to winecfg. User can double click in dlls list to edit overides of selected item.
---
 programs/winecfg/libraries.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/programs/winecfg/libraries.c b/programs/winecfg/libraries.c
index a9889e5..a98da0e 100644
--- a/programs/winecfg/libraries.c
+++ b/programs/winecfg/libraries.c
@@ -643,6 +643,18 @@ LibrariesDlgProc (HWND hDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
                         else
                             set_controls_from_selection(hDlg);
                         break;
+            case CBN_DBLCLK:
+                if (LOWORD(wParam) == IDC_DLLS_LIST)
+                {
+                    int index;
+                    POINT p;
+                    if (GetCursorPos(&p) && ScreenToClient((HWND)lParam, &p))
+                    {
+                        index = SendDlgItemMessageW(hDlg, IDC_DLLS_LIST, LB_ITEMFROMPOINT, 0, MAKELPARAM(p.x, p.y));
+                        if (!HIWORD(index)) on_edit_click(hDlg);
+                    }
+                }
+                break;
 		}
 		break;
 	}
-- 
2.3.2

